modules/bot-improvement/src/views/full/components/messages/MessageList.tsx

```diff
@@ -0,0 +1,41 @@
+import _ from 'lodash'
+import React, { FC, useEffect } from 'react'
+
+import { FlaggedMessageGroup } from '../../../../backend/typings'
+import style from '../../style.scss'
+
+import Message from './Message'
+
+export const MessageList: FC<{ messageGroups: FlaggedMessageGroup[] }> = props => {
+  const { messageGroups } = props
+  const itemRefs = {}
+
+  useEffect(() => {
+    if (_.isEmpty(messageGroups)) {
+      return
+    }
+    const flaggedMessageGroup = messageGroups.find(g => g.flagged)
+    const ref = itemRefs[flaggedMessageGroup.incoming.id]
+    messagesListRef.current.scrollTop =
+      ref.offsetTop - screen.height / 2 + ref.offsetHeight / 2 + messagesListRef.current.offsetTop
+  })
+
+  const messagesListRef = React.createRef<HTMLDivElement>()
+
+  return (
+    <div ref={messagesListRef} className={style.conversationMessages}>
+      {messageGroups.map((group, groupIdx) => (
+        <div
+          ref={el => (itemRefs[group.incoming.id] = el)}
+          key={groupIdx}
```

Comment:
Using the index as a key is not recommended, is there something else that you can use as a key ? (ex: incoming event id)